### DISCUSSION TITLE
Source Term

### DISCUSSION THREAD

**Discussion by sandip100mondal** [zwicker-group/py-pde#379] [Q&A]
Hi,
How do I add a time and space-dependent source term?

**Comment by david-zwicker** [comment]
What equation do you want to solve and what have you tried so far?

**Reply by sandip100mondal** [reply]
I want to solve a diffusion-reaction equation with a source of form d_t u(x, t) = d_x^2 u(x, t) + u(x, t) + f(x, t); here, d_t and d_x are the partial derivatives w.r.t x and t. I did not find any option to include f(x,t).

**Reply by david-zwicker** [reply]
You can simply use the [`PDE` class](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.pde.html#pde.pdes.pde.PDE) to define your PDE with a right hand side that depends on time and space.
